common/limiter/limiter.go

```diff
@@ -63,14 +73,21 @@ func (l *Limiter) AddInboundLimiter(tag string, nodeSpeedLimit uint64, userList
 		// init local store
 		gs := goCacheStore.NewGoCache(goCache.New(time.Duration(globalLimit.Expiry)*time.Second, 1*time.Minute))
 
-		// init redis store
-		rs := redisStore.NewRedis(redis.NewClient(
+		// init redis client
+		client := redis.NewClient(
 			&redis.Options{
 				Addr:     globalLimit.RedisAddr,
 				Password: globalLimit.RedisPassword,
 				DB:       globalLimit.RedisDB,
-			}),
-			store.WithExpiration(time.Duration(globalLimit.Expiry)*time.Second))
+			})
+
+		// check redis connection
+		if err := checkRedisConnection(client); err != nil {
+			return err
```

Comment:
Thanks for your PR. You directly return err and not init userMap. It could some unexpected panic. And redis is avaliable or not is not a required conditions. Wish your next PR.